kalite/control_panel/templates/control_panel/partials/_students_table.html

```diff
@@ -46,18 +46,20 @@
         <div class="table-responsive">
             <table class="table selectable-table">
                  <thead>
-                    <tr class="header-footer-bg">
-                        <th><input class="select-all" type="checkbox" value="#students" autocomplete="off"/></th>
-                        <th>{% trans "Learner Name" %}</th>
-                        <th>{% trans "Edit" %}</th>
-                        <th>{% trans "Coach" %}</th>
-                        <th>{% trans "Group" %}</th>
-                        <th>{% trans "Logins" %}</th>
-                        <th>{% trans "Login Time" %}</th>
-                        <th>{% trans "Videos Viewed" %}</th>
-                        <th>{% trans "Exercises Completed" %}</th>
-                        <th>{% trans "Mastery" %}</th>
-                    </tr>
+                    {% if student_pages %}
```

Comment:
@amodia I think you should also remove the `table` and it's container `div` if there are no students, those are additional tags on the page with nothing to do.  

I saw an [if condition](https://github.com/mrpau/ka-lite/blob/hotfix/2199-no-need-to-show-student-admin-table-header-when-no-students-exist/kalite/control_panel/templates/control_panel/partials/_students_table.html#L17) above your edits which displays the "You currently have no learner..." text.  Why don't you put the table inside the `else` block and see if that still works?